smpplib/command.py

```diff
@@ -328,9 +330,18 @@ def parse_optional_params(self, data):
 
         while pos < dlen:
             type_code, pos = unpack_short(data, pos)
-            field = get_optional_name(type_code)
             length, pos = unpack_short(data, pos)
 
+            try:
+                field = get_optional_name(type_code)
+            except exceptions.UnknownCommandError as e:
+                if self.allow_unknown_opt_params:
+                    logger.warning("Unknown optional parameter type 0x%x, skipping" % type_code)
+                    pos += length
+                    continue
+                else:
+                    raise
```

Comment:
No need to `else` here

Reply:
It still should be re-raised in case of `allow_unknown_opt_params=False`, right?

Reply:
sure
```python
if smth:
    continue
raise
```